### PR TITLE
Feature/participation requests

### PR DESCRIPTION
**Pull request by welcoMattic**
Ref #8 

For this first draft, all actions are in EasyAdmin. We will reorganize features (to separate admin concerns from user concerns) in a near future.

This feature use [Symfony Workflow](https://symfony.com/doc/current/components/workflow.html) under the wood, with the following schema:

participation_request

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,41 +12,44 @@
...
         "easycorp/easyadmin-bundle": "^2.1",
         "knpuniversity/oauth2-client-bundle": "^1.15",
         "league/oauth2-google": "^3.0",
         "myclabs/php-enum": "^1.6",
         "php-translation/symfony-bundle": "^0.5.0",
         "sensio/framework-extra-bundle": "^5.2",
         "stof/doctrine-extensions-bundle": "^1.3",
-        "symfony/asset": "^4.2",
-        "symfony/console": "^4.2",
+        "symfony/asset": "^4.3@dev",
```
Comment (lyrixx) [inline composer.json]: ca ne marche pas comme ca avec flex
https://github.com/symfony/website-skeleton/blob/master/composer.json

File: config/packages/workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            marking_store:
+                type: 'multiple_state'
+                arguments:
+                    - 'marking'
+            supports:
+                - App\Entity\Participation
+            initial_marking: [!php/const App\Enum\Workflow\Place\Participation::PENDING]
+
+            places:
+                - !php/const App\Enum\Workflow\Place\Participation::PENDING
```
Comment (lyrixx) [inline config/packages/workflow.yaml]: C'est moche :)

File: infrastructure/docker/docker-compose.yml [hunk 3]
```diff
@@ -24,19 +25,23 @@ services:
...
         volumes:
             - "../../${PROJECT_DIRECTORY}:/var/www:cached"
         labels:
             - "traefik.port=80"
             - "traefik.frontend.entryPoints=https"
             - "traefik.frontend.rule=Host:${PROJECT_HOSTNAMES}"
 
-    mysql:
-        build: services/mysql
+    database:
```
Comment (lyrixx) [inline infrastructure/docker/docker-compose.yml]: postgres

File: infrastructure/docker/docker-compose.yml [hunk 4]
```diff
@@ -2,6 +2,7 @@ version: '3'
 
 volumes:
     mysql-data: {}
+    database-data: {}
```
Comment (lyrixx) [inline infrastructure/docker/docker-compose.yml]: postgres-data

File: src/Entity/Participation.php [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+     */
+    private $needHotel = true;
+
+    /**
+     * @ORM\Column(type="boolean")
+     */
+    private $needTicket = true;
+
+    /**
+     * @ORM\Column(type="json_array")
```
Comment (Korbeil) [inline src/Entity/Participation.php]:
```suggestion
     * @ORM\Column(type="json")
```
Utiliser `type="json"`, `json_array` est déprécié

File: infrastructure/docker/docker-compose.yml [hunk 6]
```diff
@@ -2,6 +2,7 @@ version: '3'
 
 volumes:
     mysql-data: {}
```
Comment (lyrixx) [inline infrastructure/docker/docker-compose.yml]: y'a plus besoin

File: src/Repository/ParticipationRepository.php [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    public function findOneBySomeField($value): ?Participation
+    {
+        return $this->createQueryBuilder('p')
+            ->andWhere('p.exampleField = :val')
+            ->setParameter('val', $value)
+            ->getQuery()
+            ->getOneOrNullResult()
+        ;
+    }
+    */
```
Comment (Korbeil) [inline src/Repository/ParticipationRepository.php]: a clean

**Pull request merged**